app/src/main/java/com/example/instagram/fragments/ComposeFragment.java

```diff
@@ -0,0 +1,199 @@
+package com.example.instagram.fragments;
+
+import static android.app.Activity.RESULT_OK;
+
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Bundle;
+
+import androidx.annotation.Nullable;
+import androidx.core.content.FileProvider;
+import androidx.fragment.app.Fragment;
+
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import com.example.instagram.FeedActivity;
+import com.example.instagram.LoginActivity;
+import com.example.instagram.MainActivity;
+import com.example.instagram.Post;
+import com.example.instagram.R;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Use the {@link ComposeFragment#newInstance} factory method to
+ * create an instance of this fragment.
+ */
+public class ComposeFragment extends Fragment {
+    public static final String TAG = "ComposeFragment";
+    public final static int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 42;
+    private Button btnLogout;
+    private EditText etDescription;
+    private Button btnCaptureImage;
+    private ImageView ivPostImage;
+    private Button btnSubmit;
+
+    public String photoFileName = "photo.jpg";
```

Comment:
For readability, f this is immutable make it `static final`. If it's mutable do like

```
private static final String DEFAULT_PHOTO_FILE_NAME = "photo.jpg";
public String photoFileName = DEFAULT_PHOTO_FILE_NAME;
```